pkg/generator/helper/helper.go

```diff
@@ -188,35 +188,130 @@ func Test{{.CobraCmdFuncName}}(t *testing.T) {
 
 const stepGoImplementationTemplate = `package cmd
 import (
+	"fmt"
 	"github.com/SAP/jenkins-library/pkg/command"
 	"github.com/SAP/jenkins-library/pkg/log"
 	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
 )
 
+type {{.StepName}}Utils interface {
+	command.ExecRunner
+
+	FileExists(filename string) (bool, error)
+
+	// Add more methods here, or embed additional interfaces, for everything you need to be able to mock in tests.
+	// Unit tests shall be executable in parallel (not depend on global state), and don't (re-)test dependencies.
+}
+
+type {{.StepName}}UtilsBundle struct {
+	*command.Command
+	*piperutils.Files
+
+	// Embed more structs as necessary to implement methods or interfaces you add to {{.StepName}}Utils.
+	// Structs embedded in this way must each have a unique set of methods attached.
+	// If there is no struct which implements the method you need, attach the method to
+	// {{.StepName}}UtilsBundle and forward to the implementation of the dependency.
+}
+
+func new{{.StepName | title}}Utils() {{.StepName}}Utils {
+	utils := {{.StepName}}UtilsBundle{
+		Command: &command.Command{},
+		Files:   &piperutils.Files{},
+	}
+	// Reroute command output to logging framework
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
 func {{.StepName}}(config {{ .StepName }}Options, telemetryData *telemetry.CustomData{{ range $notused, $oRes := .OutputResources}}, {{ index $oRes "name" }} *{{ index $oRes "objectname" }}{{ end }}) {
-	// for command execution use Command
-	c := command.Command{}
-	// reroute command output to logging framework
-	c.Stdout(log.Writer())
-	c.Stderr(log.Writer())
+	// Utils can be used wherever the command.ExecRunner interface is expected.
+	// It can also be used for example as a mavenExecRunner.
+	utils := new{{.StepName | title}}Utils()
 
-	// for http calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// For HTTP calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
 	// and use a  &piperhttp.Client{} in a custom system
 	// Example: step checkmarxExecuteScan.go
 
-	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
-	err := run{{.StepName | title}}(&config, telemetryData, &c{{ range $notused, $oRes := .OutputResources}}, &{{ index $oRes "name" }}{{ end }})
+	// Error situations should be bubbled up until they reach the line below which will then stop execution
```

Comment:
[for the records only] Bubbling up is better compared to calling `Fatal` somewhere else in the steps. In general I think a call to a method in a logging framework should never end up in an system exit. This is simply not the job of a logging framework. So I would like have a code which comes to an end via normal code flow rather than with any system exit. `os.Exit(1)` as "last line" in main would be OK, of course.

Reply:
I do agree with that. But I would not consider it the scope of this PR. However, the step-generator would have to be changed, existing steps would all have to be adjusted, and then also the comment above would be updated.

Reply:
I guess one can like it or not. 
Many go programs rely exactly on this functionality. The standard logging library works like that as logrus does.

In the end we do benefit in a lot of places which probably would require re-inventing the wheel in many areas, e.g hooks.